### PR TITLE
build: update to clang-11

### PR DESCRIPTION
**Pull request by lizan**
Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/master/api/review_checklist.md):]

### CODE REVIEW

**Comment by moderation** [comment]
Can https://github.com/envoyproxy/envoy/blob/main/bazel/repository_locations.bzl#L824-L835 (`test_only`) and https://github.com/envoyproxy/envoy/blob/main/bazel/repository_locations.bzl#L639-L651 (WASM) be bumped at the conclusion of this PR?

**Comment by lizan** [comment]
@moderation LLVM need to be upgraded together with WAVM.

**Comment by lambdai** [comment]
@lizan What do I need to be done if I want to use machine native tool chains?

Install the clang-11, bazel/setup_clang.sh new clang11 dir and export CLANG_FORMAT=llvm-11/clang-format for sure.

What else?

**Comment by lizan** [comment]
@lambdai yes that's it.

**Comment by lizan** [comment]
@htuch this is ready to go

File: tools/clang_tools/support/BUILD.prebuilt [hunk 1]
```diff
@@ -1,7 +1,7 @@
-# Clang 10.0 library pre-built Bazel.
+# Clang 11.0 library pre-built Bazel.
 #
 # This file was mostly manually assembled (with some hacky Python scripts) from
-# clang+llvm-10.0.0-x86_64-linux-gnu-ubuntu-18.04.tar.xz and corresponding
+# clang+llvm-11.0.0-x86_64-linux-gnu-ubuntu-16.04.tar.xz and corresponding
```
Comment (htuch) [inline tools/clang_tools/support/BUILD.prebuilt:4]: Are we going backward from 18.04 to 16.04 here?

**Review by htuch — APPROVED** [review]
LGTM, thanks! @moderation can we do these bumps as distinct  followup PRs?

File: tools/clang_tools/support/BUILD.prebuilt (hunk 1 above)
Comment (lambdai) [reply tools/clang_tools/support/BUILD.prebuilt:4]: Either ubuntu 16.04 and 20.04 has prebuild 11.0.0 https://releases.llvm.org/download.html
Prebuild 11.0.0-ubuntu16.04 is running fine on my ubuntu 20.04 except lldb-11.
I guess we don't need lldb-11 during the build.

Comment (lizan) [reply tools/clang_tools/support/BUILD.prebuilt:4]: Yeah that's because they don't have build for 18.04 and 16.04 would just work fine in 18.04.

Comment (htuch) [reply tools/clang_tools/support/BUILD.prebuilt:4]: Fair enough; prefer going forward than back all things equal but not if there are complications.

**Comment by lizan** [comment]
The clang-tidy error can be ignored. merging this.

**Pull request merged**